src/npc_attack.cpp

```diff
@@ -105,8 +140,7 @@ void npc_attack_melee::use( npc &source, const tripoint &location ) const
 
 tripoint_range<tripoint> npc_attack_melee::targetable_points( const npc &source ) const
 {
-    const int reach_range{ weapon.reach_range( source ) };
-    return get_map().points_in_radius( source.pos(), reach_range );
+    return get_map().points_in_radius( source.pos(), 8 );
```

Comment:
an npc can't technically target these points. explain?

Reply:
moving toward enemies is in a different part of the code. this function is "the npc has already chosen to use melee, what are the targets?"

Reply:
It doesn't work that way though. As long as the NPC has any items it will decide throwing said items is the way to go and won't bother moving into melee range.